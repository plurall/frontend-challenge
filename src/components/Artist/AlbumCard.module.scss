@import 'shared.module.scss';

.wrapper {
  font-family: 'Roboto', sans-serif;
  background: var(--color-secondary-surface);
  border-radius: 0.5rem;
  padding: 0.625rem 2rem;
  user-select: none;
  text-decoration: none;
  cursor: pointer;

  display: grid;
  grid-template-columns: 1fr auto;
  grid-template-rows: auto 1fr auto;
  grid-template-areas:
    'name picture'
    'details picture'
    'button picture';
  transition-duration: 0.25s;

  &:hover {
    box-shadow: var(--shadow-artist-card-hover);
  }

  @media (max-width: $mobile-breakingpoint) {
    padding: 0.625rem 1rem;
  }
}

.name {
  grid-area: name;
  color: var(--color-on-surface-highlight);
  font-weight: 700;
  font-size: 1.375rem;

  @media (max-width: $mobile-breakingpoint) {
    font-size: 1.125rem;
    font-weight: 600;
  }
}

.details {
  grid-area: details;
  font-size: 0.875rem;
  color: var(--color-on-surface);

  display: flex;
  align-items: center;
}

.date {
  font-weight: 600;
}

.separator {
  display: inline-block;
  width: 0.25rem;
  height: 0.25rem;
  border-radius: 50%;
  background-color: var(--color-on-surface);
  margin: 0 0.375rem;
}

.tracks {
  font-weight: 500;
}

.button {
  width: 2rem;
  height: 2rem;
  border-radius: 50%;
  background-color: var(--color-primary);
  color: var(--color-on-primary);
  font-size: 1.5rem;
  border: none;

  display: flex;
  align-items: center;
  justify-content: center;

  & > * {
    transform: translateX(5%);
  }
}
