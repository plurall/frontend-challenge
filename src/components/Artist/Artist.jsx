import PropTypes from 'prop-types'
import React, { Component } from 'react'

import styles from './Artist.module.css'

class Artist extends Component {
  render() {
    console.log(this.props.artist);

    let image = null;
    if (this.props.artist.images[0]) {
      image = <img src={this.props.artist.images[0].url} width="120px" height="120px" alt="artist image"/>;
    } else {
      image = <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAOEAAADhCAMAAAAJbSJIAAAAY1BMVEX///9oaGhjY2NeXl5hYWFaWlrOzs5fX19YWFienp55eXnk5OT7+/uDg4OTk5P19fWfn5+/v7/q6urU1NSNjY23t7fIyMjc3NympqZra2uvr6+QkJCGhoa5ublzc3PZ2dlQUFC1P+9nAAAOHklEQVR4nO1d6dKquhLdZhBBBhVRnM/7P+UFP9EkpJMOhOitYlWdP2f7QRbp9JRO59+/GTNmzJgxY8aMGTNmzJgxY4YnpLvH4XbexJfrNcuy6z6P62q7PO6+PS4f2JVVnESc0ogxIoIxSjlfXOvtsfj2IIcifZyvDTdGFgY0VClf5Lfjt0frjOM5s5ETeTY0r9X624PGYxkzPLsPS07i8ttDx6DMOXVl92ZJef7jJHc1G0yvI8nq3xXXZcLH0XuR5Mnh21R0SCsycvoEUHZOv01IQVFz5oveE4zXv2QnG37epu8Nstr8DMfzBPz+ONY/IavbyCqfhEWNi8bJ6Z60uDcyyHnjyFk/DKO3b9P7d7xTM7eGWhJXh+NOno6iccbPedIQJUae0em7BjLNV2Z213NpjiB2S5t/x/dfXI4H2D407kmG9jTXt6vBESJ8OykLGMUVElBCo7h0VBJlDLtDNPvKNIITGA11Lh8xBZ5J+Be8nJwDYxnlch0ywPLwi7eR47A7aQdCVvlYt3m30XtHZBE06XHQqlBPnlZ6ZlqOq4CSutFJKKP+vOVKy5HHvp5vQxbp5NOvh6X1BKPM5ytAFLolyC++9Xkaa1YCOQUwG2uNQo/uU+TK1knf3pJo8gTAo78ECa8metlWI6p84sTjsi86NJlOjRdZfxpXk7riGiuxmmoC/3DTvHE53ev6BNlpaju8XvQkdTqKfRHl+VTvErDvSepUglr2CK7ChDXn/osnUTdr9T2EhUrdaoRngsWxU2WF3MMFbeueE8e8vzztvSKMB/XCUnUUycn3KxJFo9EQOuaNtca78fyFY0VMaDA3v8VaF8tEG5+v2CqvoLXPp9uw06cTfGY2VDUamCCU8Fr5U6jKk6lX+bABJNjA1zsuspaJgioZE0HiKT11kJcBufp5LA4mgr6WYqEQvPt4KBZmgg1FH4Y/k2WUhNzwshH0YhUP8kumcAhBWAk2Wm+0818oBENm1xEEPcjpXpJRFtKVQRFckP24t5SSmgmqZUCCRPYg+bhw+CQ/LOAihAlmivkaFWVUUtAyflXjARNM1DggGpELS6XXhDT1hhls/1kRreEGrJbUjBfrioOFoBIKkMGBQCE9J6CMggTfiYVaktPV0G8fS1OY+Bq/FXaCipySgUZMnsJVsJpIDEFl/2TgJG7EKSSjQqbd+liWy8NhK+JwOCyXZQPZBqEIKq4IGRSxyjHFGDVTJCvaIooi1iF64vm/Kb8LHJEEVQkbMryzuJjZiLxF2t95UEA+3w9LUFE27DxgYJK2oiNiptpeefpeA3iCirGm7uOSoib7J3pUedLMVbKvD6rAYBzoVWomqIsDK/HLUfegR0oB272G/17Fd4SsFIHWZHP7oKUzQeXTORszKUeJcBoo+OODpnKjB9bWkWoTv89H6ocvaQruas0khw2hqT6DU5XSDVMA3n4WV4L/UmkWXA2GOCyMLbyc2rLf5j+WKN7dGcUwdyeoWGxHXVOKMo4UgGK33hX9BVtjisDJHiYIh93SUqJue9+5OKxxkf0GVeZ+B39leruYB3T0usSPE9mDimb6GhwbrGU0/2OPYgiCmA4JLSVRcyEoC6nNVJxJswRBjD2pQDiLweyJOE4nMd04zH568ntaRkcSLNqRBuoSQ0k6ypLLUjdPJwFUfHGUPC88QUmv2bRwAH7N4KFJlOYCb/RFI22de0jLewU4CtEWgZ+hj6vwZzYhLcIwhDLbD0FMHbZpXHRwIIZgJnOQvRCjAWuS1JlhexafmA87uTAUzS3FFoNthWXIbIfHXBgSxmmS1+dzHe9PnDooYXj/RRosdiGKLptVP6VohmyVbUXTvT7gJxI2yqLeR+/ri4+2e+xIhpRUagyGiqteY4djI3EhEhzBVMhiIXbneBfcP6EbXbPydGeFHoaTfe/nvR5oSKOICxGZERT9BIRkJ3/I/pD0kWWXSuNYFmDKgtPTPcmuLdoH3hucYHUgxp/0gWIorl2bNRyOApo85lgZL0YJDLe3UrtPuzt65ZTvQbpWAYnKHJnKED2ayH3sFqTrclvvF6CIum9Suns1C+e/QKPcNGaQMoO9H7BlnYgDRv2F+6wjsV1Yw+Ehxc3iJiBqO1hM71g9mgZ5rMWmQV1vPm98nBCp4fvlkisPzFvEBhUi5r5RtRSiscAkBlYExupjxjZG6/eRGj1MBZ9iyhnlmYrZHcwfmHyaT55Md2bRAaas+8NxSiRziJl0A8PPOTM4WSiMzsTQMIfSssIYREmsEQsXZvjR/Gqlv+7Hp+M/2E81MRQNImofUTL4iN/DDN9TmNtd7Kh1gCvwd0alLip/zE6uqHwR5gWOnt77DUt7+MFyy7NM2SKXrFILKaE/hmHUhROIPeBXCBND0mwcufg7jM8nRiMIDwNm2Ik4YvOpW7EP0JkzMRRqa1C1mKJbitiTARl2YTkqCdDtUEI2xcjw3v9URmRuDME8TSekW0wk32n5HBBT49aC4JiiHGlfDLvA6w78u8wgM3+On2T40lIFzlt7iSlU1mBkKEppQIadVkMVKrzdLWjRohli1uHVTZeCDF/eBWqT+2Oqh6xDV116cbOHEMNOd1yQDF8fPwH+2cRQ/B3GHopWF1HfAMlV5+VnyLTvS1yu+p8bGQrrAOXTOPqlIMPSjeHrYwK7/kaGrn6p6IIgUm02hoDYeWToHFs4xoc2KcVua7ycfEhKDevLOT6UkgL2HDLE8OXSoDyaFi/DBGkag450TbtI3m9kr2mEGP7JC5i671F4TRJkLQwMnfM0OzexNtpoMHXfw+tN4NMMDJ1zbeJLMMoXWmgnXbMHEK/1cIS8NgND53ypJCiIulTQ80413XNghn8Pg7YUTQwzt6SE8hcIgwgqy5NDArETFkCVGhm671uIlVQIufZSi/Eq8ITDaZjhgL0nUVIQ2tcHw+7bg4GIwR4O2D8UzQVCmfpg2B05Ah0gg08zYA9YmnZ7vOWBYXe8+AhnE2GGA/bx5Wq4EAy7MBTSM0aGkn3BEZRCOnu933iGnTrT9POzM5TqabCF0JVLTdR4hu/aUUPOCh67UwFXh6NLXdtohrzTDqbMMcxQqqNENzkTB221+eMYfprnHk0eEMzQaaxviF6NtaIGX9emGfjqPfLC6AGBDMth9aXVBDXCJLmdlB7IhCdvsUr1rZetDEW326FGWKrztpXU4Biy1odfn5O2w/wTEWfxZ9mklk1ikOHQw13SyT6LmKIYRp3CSh+3eJ9k+3orDmdnO2YKSdJjQGXiE9IuqUVMMQwtraVKayYAGoTLQOFXWjQUgqHl3YhCFOgJ0rEJpypDaSGakzWYOSSGVtilTskiGUqxiNO5J2kfDz7/1wJXBU2gdu2PBPf3eoZSSZvb2TX8+UNsnXdj+nqhTbG9I6+o0Ye26+FCqpwhNS0jfK0+oSw+vHMGxaPK8Dco6RlKhQ0Op56eQJ8Ddjpv0VhBTpLrNbtb7w1EMJQ2X51buEgHUE2R/oAzM87nSQCG0iw4n+WWC7UMcTDQetMzdAzlo9zuzWWwPRUCMdTIoBRtDeipgO2LgWqZMJ5hfw7lvhiueqaFVEYH9zYJw5D3s5rje5vIhg7MYjkc7RkOTeNpuYplWLsvSVVBHkO/Ufv7L+Dj3W6I6EpjkKUKiIHdvpQuPtocSK9R+4dfXS794LDUyI+cWh3a7EvuhqCrHjI0zJm4uZuUlxt8aELp19bPEeygNchGNqS0Qm53OLhfm9L/qLcPVUCuyeQEZUdqRBMrpW+i4jZ8j6BSGT+midVN7n0pyWkBeZds8g6SlfTlo1H3XMpNJsW+e+AM+j4N1sdOVuDjbkkoFa/z8xZIyUxPUO3tOfIQqFyF9d5sduzK5RW5fkhDoYR/9M//A53RAASVexrGn3LdymSe1/TAPZ2mJ6hs4PjoqqqUTzZW8ZsElTIyL2/spSnAQz4BCP5TNnD8nMRWLygBK0ICEFQFylOLeFy3rhCXBymH4Hzdb4FrlBSCYKyqcG9PBqPAsARrhaDP1sbqUvwJgn4vs7a0LAtBcKMQ9H1Pg/EkbwgtmisEB6SAzejduxaY4LVno7zfBgPfp+H9a/bRL2SgE1xEAflqAQiue8H2NB3i9V2PAtxb0r/+dKprOjVX5S4W98lvR4p7wjPdVauay3KRdcfDsbv3DNWUl6xrKE581aPuPuBJb2TSCCo7TXcvhPZO5wkvPG6h24lRu7B7Q6Wp1Jj20uoWunM7bDHFa493zascurAOhrbKju9978QUF439/dTbToo00dbCGgq7BryjXmm+I0tCXdyX69wbRofsNGuRnrWlUkFuAX9Bfyqt4ehDVotaXysV6BbwF/qe4h9HHo/VBOtcJ5/tEgx2WdEf0qs+1iC6Pp54HBKglI9PvmPXxxYYC4l4jDtgpaLMOZCpJEFvlnyjSMDdCxrFrhayjBnYc5BOXRUA4gZXhxLKrzfsyllv9xxuqUhC3kqoorgaknCEUZ6dl2YbXTyqfcRNhZj+nQk3lObTvqSZS57k58NjLY8z3R2Xt82V2MpM6WnYmvaJyn6DBWERbZhGi/uzsfD9RNpLaYx9TP/A6Kg9el/Qe1ge4NkTHINiMwFHwjffXYAyIE9rMBh0fOF7SM+wPXMFoaT6FfmUALpcjvzGOX7TYleTkRNJKKtDXm4+AI1zOZhkYzzzydMwPlDGxOil6NkxTpyd2S9iXWUOh2Ke/l0VOP7zgGN1eTplxp3HlhzZV+hj5j+H4nHbZM9LyxiTGnSzxpHjnGXxrfw1uzcErZu9reo4z/f763W/z+P6vD08dj9p82bMmDFjxowZM2bMmDFjxoz/T/wPcKqtaCSHCKsAAAAASUVORK5CYII=" width="120px" height="120px" alt="artist no image"/>;
    }

    return (
      <div className={styles.artist}>
        <div>
          {image}
        </div>
        <div className={styles.inner}>
          <h2>{this.props.artist.name}</h2>
        </div>
      </div>
    )
  }
}

export default Artist
