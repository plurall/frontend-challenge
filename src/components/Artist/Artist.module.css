
.artist {
  padding: 10px;
  margin: 15px 0;
  background: #fff;
  border-left: 2px solid #655AA3;
  display: flex;
  align-items: center;
}

.artist:hover {
  background: #655AA3;
  border-left: 2px solid #fff;
}

.artist .inner {
  padding-left: 20px;
}

.artist h2 {
  padding: 15px 0;
  text-decoration: none;
  color: #000;
}
