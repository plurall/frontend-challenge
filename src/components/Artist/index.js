export { default as ArtistDetailedCard } from './ArtistDetailedCard'
export { default as AlbumCard } from './AlbumCard'
export { default as AlbumsList } from './AlbumsList'
export { default as LoadingArtistDetailedCard } from './LoadingArtistDetailedCard'
export { default as LoadingAlbumsList } from './LoadingAlbumsList'
export { default as LoadingAlbumCard } from './LoadingAlbumCard'
