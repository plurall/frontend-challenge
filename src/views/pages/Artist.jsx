import React, { useEffect, useState } from 'react'
import { format } from 'date-fns'
import { utcToZonedTime } from 'date-fns-tz'
import { SomosClient } from 'utils'
import { useParams } from 'react-router-dom'
import Breadcrumb from 'components/Breadcrumb/Breadcrumb'

import styles from './Artist.module.css'

const client = new SomosClient()
const Artist = () => {
  const { id } = useParams()

  const [artist, setArtist] = useState()
  const [albums, setAlbums] = useState()

  const routes = [
    { name: 'Home', url: '/' },
    { name: 'Busca', url: '/busca' },
    { name: 'Artista', url: `/artista/${id}` },
  ]

  const name = 'Nome: '
  const genre = 'Gênero: '
  const popularity = 'Popularidade: '

  useEffect(() => {
    async function getArtist() {
      const data = await client.getArtistId(id)
      const dataAlbums = await client.getAlbumsById(id)
      setArtist(data)
      setAlbums(dataAlbums.items)
    }
    getArtist()
  }, [id])
  return (
    <>
      <Breadcrumb routes={routes} />
      <h1>Artista</h1>
      <div className={styles.content}>
        <div className={styles.card}>
          <div>
            <img className={styles.imgArtist} alt={artist?.name} src={artist?.images[0]?.url} />
          </div>
          <div className={styles.infos}>
            <label htmlFor="name">{name}{artist?.name}
            </label>
            <label>{genre}
              {artist?.genres.join(', ')}
            </label>
            <label>{popularity}
              {artist?.popularity}
            </label>
          </div>
        </div>
        {albums?.length && (
          <div className={styles.section}>
            <h1>Álbuns</h1>
            <div className={styles.albums}>
              {albums.map(album => (
                <div className={styles.albumInfo}>
                  <img className={styles.img} alt={album.name} src={album.images[0]?.url} />
                  <div className={styles.albumName}>
                    {album.name}
                  </div>
                  <div className={styles.data}>
                    {format(utcToZonedTime(album.release_date, 'UTC'), 'dd/MM/yyyy')}
                  </div>
                </div>
              ))}
            </div>
          </div>
        )}
      </div>
    </>
  )
}


export default Artist
