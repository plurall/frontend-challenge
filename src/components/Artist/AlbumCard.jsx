import React from 'react'
import { Link } from 'react-router-dom'
import { BsPlayFill } from 'react-icons/bs'

import { albumType } from 'types'
import { formatDate } from 'utils'
import { ArtistPicture } from 'components'
import styles from './AlbumCard.module.scss'

const AlbumCard = ({ album }) => {
  const pathname = `${process.env.REACT_APP_ALBUM_EXTERNAL_URL}/${album.id}`

  return (
    <Link to={{ pathname }} target="_blank" className={styles.wrapper}>
      <h3 className={styles.name}>{album.name}</h3>
      <p className={styles.details}>
        <span className={styles.date}>{formatDate(album.release_date)}</span>
        <span className={styles.separator} />
        <span className={styles.tracks}>{album.total_tracks} faixas</span>
      </p>
      <button className={styles.button}>
        <BsPlayFill />
      </button>
      <ArtistPicture src={album.image} />
    </Link>
  )
}

AlbumCard.propTypes = {
  album: albumType.isRequired,
}

export default AlbumCard
