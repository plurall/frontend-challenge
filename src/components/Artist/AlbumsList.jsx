import React from 'react'
import PropTypes from 'prop-types'

import { albumType } from 'types'
import { AlbumCard } from 'components'

import styles from './AlbumsList.module.scss'

const AlbumsList = ({ albums, total, show }) =>
  show && (
    <div className={styles.wrapper}>
      <div className={styles.header}>
        <h2 className={styles.title}>Álbums</h2>
        <p className={styles.totalCount}>{total} álbums relacionados</p>
      </div>
      <ul className={styles.list}>
        {albums.map(album => (
          <AlbumCard key={album.id} album={album} />
        ))}
      </ul>
    </div>
  )

AlbumsList.propTypes = {
  albums: PropTypes.arrayOf(albumType).isRequired,
  total: PropTypes.number.isRequired,
  show: PropTypes.bool.isRequired,
}

export default AlbumsList
