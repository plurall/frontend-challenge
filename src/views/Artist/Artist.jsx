import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";

import { SomosClient } from "utils";
import { CardGrid, Layout } from "components";

import styles from "./Artist.module.css"

const Artist = ({ match: { params: { id } } }) => {
  const [artist, setArtist] = useState({});
  const [albums, setAlbums] = useState([]);

  const client = new SomosClient();

  async function handleGetArtist(id) {
    const fetchedArtist = await client.getArtist(id);

    if(!fetchedArtist.id) return;
    
    const newArtist = {
      id: fetchedArtist.id,
      name: fetchedArtist.name,
      popularity: fetchedArtist.popularity,
      image: fetchedArtist.images[0] ? fetchedArtist.images[0].url : 'images/default-icon.png',
      genres: fetchedArtist.genres.join(', ')
    }
    
    return newArtist
  }

  async function handleGetArtistAlbums(id) {
    const fetchedAlbums = await client.getArtistAlbums(id);

    if(!fetchedAlbums.items) return;

    const newAlbums = fetchedAlbums.items.map(album => {
      return {
        id: album.id,
        name: album.name,
        date: new Date(album.release_date).toLocaleDateString("pt-BR"),
        image: album.images[0] ? album.images[0].url : 'images/default-icon.png',
      }
    })

    return newAlbums;
  }

  useEffect(() => {
    handleGetArtist(id).then(artist => setArtist(artist));
    handleGetArtistAlbums(id).then(albums => setAlbums(albums))
  },[])

  return (
    <Layout>
      <div className={ styles.container } >
        <div className={ styles.info } >
          <img src={ artist.image } alt="icon" />
          <div>
            <span className={ styles.name } >{ artist.name }</span>
            <span className={ styles.popularity } > <strong>Popularidade:</strong> { artist.popularity }</span>
            <span className={ styles.genres } > <strong>Gêneros:</strong> { artist.genres }</span>
          </div>
        </div>
        <div className={ styles.albums } >
          <p>Álbuns</p>
          <CardGrid elements={ albums } />
        </div>
      </div>
    </Layout>
  )
}

export default Artist