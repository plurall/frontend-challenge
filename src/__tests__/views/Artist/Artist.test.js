
import React from 'react';
import { shallow } from 'enzyme';
import Artist from 'views/Artist';

let wrapper;
const props = {
  match:{
    params:{
      id: 1
    }
  }
};

beforeEach(() => {
  wrapper = shallow(<Artist {...props} />);
});


describe('[Tela Artista]', () => {

  it('deve ser definido o componente', () => {

    expect(Artist).toBeDefined();
  });

  it('deve equivaler ao snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

  it('deve chamar a função getArtistInfo no `componentDidMount()`', () => {
    const instance = wrapper.instance();
    jest.spyOn(instance, 'getArtistInfo');
    instance.componentDidMount();

    expect(instance.getArtistInfo).toHaveBeenCalledTimes(1);
  });

  it('deve chamar a função getArtistAlbums no `componentDidMount()`', () => {
    const instance = wrapper.instance();
    jest.spyOn(instance, 'getArtistAlbums');
    instance.componentDidMount();

    expect(instance.getArtistAlbums).toHaveBeenCalledTimes(1);
  });

});

