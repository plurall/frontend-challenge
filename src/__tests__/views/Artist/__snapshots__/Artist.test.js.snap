// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Tela Artista] deve equivaler ao snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Artist
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <SubHeader
          breadcrumb={
            Array [
              Object {
                "text": "Home / Artist",
              },
            ]
          }
          heading="Somos Front-end Challange"
        />,
        <div
          className="wrapper"
        >
          <Button
            href="/busca"
            text="Voltar"
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "breadcrumb": Array [
            Object {
              "text": "Home / Artist",
            },
          ],
          "heading": "Somos Front-end Challange",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [],
            <Button
              href="/busca"
              text="Voltar"
            />,
          ],
          "className": "wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "href": "/busca",
              "text": "Voltar",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <SubHeader
            breadcrumb={
              Array [
                Object {
                  "text": "Home / Artist",
                },
              ]
            }
            heading="Somos Front-end Challange"
          />,
          <div
            className="wrapper"
          >
            <Button
              href="/busca"
              text="Voltar"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "breadcrumb": Array [
              Object {
                "text": "Home / Artist",
              },
            ],
            "heading": "Somos Front-end Challange",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [],
              <Button
                href="/busca"
                text="Voltar"
              />,
            ],
            "className": "wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "href": "/busca",
                "text": "Voltar",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
